src/drive/mobile/modules/authorization/DriveMobileRouter.jsx

```diff
@@ -1,108 +1,93 @@
-/* global cozy */
-
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 import Proptypes from 'prop-types'
-import MobileRouter from 'authentication/MobileRouter'
+import localForage from 'localforage'
+import { MobileRouter } from 'cozy-authentication'
+import { getUniversalLinkDomain } from 'cozy-ui/transpiled/react/AppLinker'
+import { withClient } from 'cozy-client'
 import AppRoute from 'drive/web/modules/navigation/AppRoute'
 import { setToken } from 'drive/mobile/modules/authorization/duck'
 import { setUrl } from 'drive/mobile/modules/settings/duck'
 import { restoreCozyClientJs, initBar } from 'drive/mobile/lib/cozy-helper'
 import { IconSprite } from 'cozy-ui/transpiled/react/'
-import {
-  unlink,
-  isAuthorized,
-  isRevoked,
-  getOnboardingInformations
-} from './duck/index'
+
+import { unlink } from './duck/index'
 import { saveCredentials } from './sagas'
 import { setCozyUrl } from 'drive/lib/reporter'
+import { ONBOARDED_ITEM } from 'drive/mobile/modules/onboarding/OnBoarding'
+import { PROTOCOL, SOFTWARE_NAME } from 'drive/mobile/lib/constants'
+import appMetadata from 'drive/appMetadata'
 
-import {
-  onboardingInformationsPropTypes,
-  onboardingPropTypes
-} from '../../../../authentication/OnboardingPropTypes'
+import appBooted from '../../../targets/mobile/'
 class DriveMobileRouter extends Component {
-  static contextTypes = {
-    client: Proptypes.object.isRequired
+  async componentDidMount() {
+    //Wait for the app to be booted to avoid race condition between cordova & JS
+    await appBooted
```

Comment:
I'm not sure how that works, I thought awaiting in componentDidMount didn't prevent the initial rendering of the component..?

Reply:
Oh you have totally right! changed to use a state instead